src/main/java/teammates/storage/sqlentity/UsageStatistics.java

```diff
@@ -0,0 +1,157 @@
+package teammates.storage.sqlentity;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.annotations.UpdateTimestamp;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+
+/**
+ * Represents a system usage statistics for a specified period of time.
+ *
+ * <p>Note that "system usage" here is defined as user-facing usages, such as number of entities created
+ * and number of actions, as opposed to system resources such as hardware and network.
+ */
+@Entity
+@Table(name = "UsageStatistics")
+public class UsageStatistics extends BaseEntity {
+    @Id
+    private String id;
+
+    @Column
+    private Instant startTime;
+
+    @Column
+    private int timePeriod;
+    @Column
+    private int numResponses;
+    @Column
+    private int numCourses;
+    @Column
+    private int numStudents;
+    @Column
+    private int numInstructors;
+    @Column
+    private int numAccountRequests;
+    @Column
+    private int numEmails;
+    @Column
+    private int numSubmissions;
+
+    @CreationTimestamp
+    @Column(updatable = false)
+    private Instant createdAt;
+
+    @UpdateTimestamp
+    @Column
+    private Instant updatedAt;
+
+    @Column
+    private Instant deletedAt;
```

Comment:
All these fields should be not null according to our data model design document. If `nullable` is not specified for the column annotation, the default is true.

Reply:
Can primitive type `int` have null value? If it still can have null value I will add `nullable = false`

Reply:
Don't confuse `@Column(nullable = false)` with other similar `@NotNull` definitions. The column annotation is not for the java class, it's for the table column in the db. Hibernate uses these to add NOT NULL constraint to the db column for cases when it generates the DDL.

Reply:
ok I see. Do we still need `@Column(nullable = false)`? considering that its impossible to have these attributes as null from the java code

Reply:
we will be interacting with the db through other means as well (e.g. migrations). there's a discussion to be had about whether we need these annotations (they are only used when hibernate generates DDL), but a discussion for another day.